firefox-ios/Providers/Profile.swift

```diff
@@ -218,6 +219,44 @@ extension Profile {
     }
 }
 
+func remoteSettingsAppContext() -> RemoteSettingsContext {
+    let encoder = JSONEncoder()
+    let appInfo = BrowserKitInformation.shared
+    let uiDevice = UIDevice.current
+    let formFactor = switch uiDevice.userInterfaceIdiom {
+    case .pad: "tablet"
+    case .mac: "desktop"
+    default: "phone"
+    }
+    var customTargetingAttributes: String?
+    let customTargetingAttributesData = try? encoder.encode([
+            "form_factor": formFactor,
+            "country": Locale.current.regionCode,
+    ])
+    if let data = customTargetingAttributesData {
+        customTargetingAttributes = String(
+            decoding: data,
+            as: UTF8.self)
+    }
+    return RemoteSettingsContext(
+        appName: "Firefox iOS",
+        appId: "org.mozilla.ios.Firefox",
+        channel: appInfo.buildChannel?.rawValue ?? "release",
+        appVersion: AppInfo.appVersion,
+        appBuild: AppInfo.buildNumber,
+        architecture: nil,
+        deviceManufacturer: "Apple",
+        deviceModel: DeviceInfo.deviceModel(),
+        locale: Locale.current.identifier,
+        os: "iOS",
+        osVersion: uiDevice.systemVersion,
+        androidSdkVersion: nil,
+        debugTag: nil,
+        installationDate: nil,
+        homeDirectory: nil,
+        customTargetingAttributes: customTargetingAttributes)
+}
+
```

Comment:
Is there a better place to put this function?  I tried adding it to `RemoteSettingsUtils.swift`, but I couldn't figure out how to properly import it here.

Reply:
Heya. So yeah, in general, we prefer not having free functions in Swift/our app.
I'm not sure why it wouldn't work. Have you pulled the latest from `main` done a clean and rebuild of the app?
I literally copy & pasted your function into the `RemoteSettingsUtils` class, without doing anything else, and the build was successful:

<img width="956" alt="Screenshot 2025-03-13 at 13 15 42" src="https://github.com/user-attachments/assets/e4072d2b-e581-4e9d-ab0f-ac86e6c63ea3" />

Reply:
That makes sense.  I was trying to add a top-level function there and failing, let's see what happens with a method.

Reply:
This seems like the right approach, but I'm not sure how to import the items I need.